src/main/java/rims/command/DeleteCommand.java

```diff
@@ -0,0 +1,46 @@
+package rims.command;
+
+import rims.core.ResourceList;
+import rims.core.Storage;
+import rims.core.Ui;
+import rims.exception.RimException;
+
+public class DeleteCommand extends Command {
+    protected String resourceName;
+    protected int qty;
+    protected char type;
+
+    public DeleteCommand(String itemName, int qty) {
+        this.resourceName = itemName;
+        this.qty = qty;
+    }
+
+    public DeleteCommand(String roomName) {
+        this.resourceName = roomName;
+        this.qty = 1;
+    }
+
+    @Override
+    public void execute(Ui ui, Storage storage, ResourceList resources) throws Exception {
+
+        int availableResourceQty = resources.getAvailableQuantity(resourceName);
+        if (availableResourceQty < qty) {
+            throw new RimException("Insufficient available resources to delete!"); //Resources not booked not enough to delete
```

Comment:
Good use of exception here. However, remember to keep your lines short (under 110 character would be good!). In this case, your comment is redundant since your exception message is quite explicit.